qc_schema/dev/definitions.py

```diff
@@ -39,3 +39,183 @@
+        },
+        "shell_region": {
+            "description": "The region this shell describes",
+            "type": "string",
+            "enum": [ "valence", "polarization", "core", "tight", "diffuse" ]
+        },
+        "shell_angular_momentum": {
+            "description": "Angular momentum (as an array of integers)",
+            "type": "array",
+            "minItems": 1,
+            "uniqueItems": True,
+            "items": {
+                "type": "integer",
+                "minimum": 0
+            }
+
+        },
+        "shell_exponents": {
+            "description": "Exponents for this contracted shell",
+            "type": "array",
+            "minItems": 1,
+            "items": {
+                "type": "string"
+            }
+        },
+        "shell_coefficients": {
+            "description": "General contraction coefficients for this contracted shell",
+            "type": "array",
+            "minItems": 1,
+            "items": {
+                "description": "Segmented contraction coefficients",
+                "type": "array",
+                "minItems": 1,
+                "items": { "type": "string" }
+            }
+        }
+    }
+}
+
+definitions["basis_ecp_potential"] = {
+    "description": "ECP potential",
+    "additionalProperties": False,
+    "required": [
+        "potential_ecp_type",
+        "potential_angular_momentum",
+        "potential_r_exponents",
+        "potential_gaussian_exponents",
+        "potential_coefficients"
+    ],
+    "properties": {
+        "potential_ecp_type": {
+            "description": "Type of the ECP Potential",
+            "type": "string",
+            "enum": [ "scalar", "spinorbit" ]
+        },
+        "potential_angular_momentum": {
+            "description": "Angular momentum (as an array of integers)",
+            "type": "array",
+            "minItems": 1,
+            "uniqueItems": True,
+            "items": {
+                "type": "integer",
+                "minimum": 0
+            }
+        },
+        "potential_r_exponents": {
+            "description": "Exponents of the r term",
+            "type": "array",
+            "minItems": 1,
+            "items": {
+                "type": "integer"
+            }
+        },
+        "potential_gaussian_exponents": {
+            "description": "Exponents of the gaussian term",
+            "type": "array",
+            "minItems": 1,
+            "items": {
+                "type": "string"
+            }
+        },
+        "potential_coefficients": {
+            "description": "General contraction coefficients for this contracted shell",
+            "type": "array",
+            "minItems": 1,
+            "items": {
+                "description": "Segmented contraction coefficients",
+                "type": "array",
+                "minItems": 1,
+                "items": { "type": "string" }
+            }
+        }
+    }
+}
+
+
```

Comment:
nit: one blank line too many?

Reply:
Removed (I imagine yapf will be run at some point)